src/mem_scan.c

```diff
@@ -119,10 +119,20 @@ void scan_mem(sqlite3 *db, int pid)
 
 	/* Maintain the list of paths that have already been had their ELF parsed, so that
 	 * we don't duplicate data or scan unnecessarily.
-	 * Choosing 50 as the max number for now, ideally it should be able to grow with the 
-	 * number of paths identified.
 	 */
-	struct kinfo_vmentry *seen_kivp = calloc(200, sizeof(struct kinfo_vmentry));
+	size_t seen_kivp_buffer_size;
+	size_t seen_kivp_buffer_capacity;
+	struct kinfo_vmentry *seen_kivp;
+
+	const int initial_size = 100;
```

Comment:
This should likely be a power of two if you're going to double using shift and not `*= 2;`

Reply:
Whoops, didn't think it through. Yes, I did mean *=2, thanks!